kong/globalpatches.lua

```diff
@@ -288,35 +288,36 @@ return function(options)
     local old_tcp_connect
     local old_udp_setpeername
 
-    local function tcp_resolve_connect(sock, host, port, sock_opts)
-      local target_ip, target_port = toip(host, port)
-      if not target_ip then
-        return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
+  -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
+    local function strip_nils(port, opts)
+      if port and opts then
+        return port, opts
+      elseif opts then
+        return opts
+      elseif port then
+        return port
       end
-
-      -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
-      if not sock_opts then
-        return old_tcp_connect(sock, target_ip, target_port)
-      end
-
-      return old_tcp_connect(sock, target_ip, target_port, sock_opts)
     end
 
-    local function udp_resolve_setpeername(sock, host, port)
+    local function resolve_connect(f, sock, host, port, sock_opts)
       local target_ip, target_port
-
       if sub(host, 1, 5) == "unix:" then
-        target_ip = host -- unix domain socket, so just maintain the named values
-
+        target_ip = host
```

Comment:
please do not remove the vertical whitspace, see also [style guide](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#conditional-expressions)

Reply:
like this one right?

```lua
    local function resolve_connect(f, sock, host, port, sock_opts)
      local target_ip, target_port
      if sub(host, 1, 5) == "unix:" then
        target_ip = host
        
      else
        target_ip, target_port = toip(host, port)
        if not target_ip then
          return nil, "[toip() name lookup failed]: " .. tostring(target_port)
        end

      end

      return f(sock, target_ip, strip_nils(target_port, sock_opts))
    end
```

Reply:
This is fine, no need for the blank line between the two `end` statements. In general, check the rest of the code.